### PR TITLE
Allow value `-1` for `transfers` option in `journeys()`

### LINKED ISSUE #5: Allow value `-1` for `transfers` option in `journeys()`
In hafas-client, it was possible to explicitly leave the amout of transfers for `journeys()` unrestricted by setting the `transfers` option to `-1` (see https://github.com/public-transport/hafas-client/blob/main/docs/journeys.md). However, this does not work for both profiles here.

http://localhost:3000/journeys?from=8000261&to=8011160&transfers=-1 returns
```json
{"message":"Unprocessable Entity","url":"https://int.bahn.de/web/api/angebote/fahrplan"}
```
for `db` and
```json
{"message":"Bad Request","url":"https://app.vendo.noncd.db.de/mob/angebote/fahrplan"}
```
for `dbnav`

A simple ternary expression in https://github.com/public-transport/db-vendo-client/blob/8026689ee89c6a8eb9223a18155b0277b85c631b/p/db/journeys-req.js#L10 and in https://github.com/public-transport/db-vendo-client/blob/de78b40da4f014ae1304d1f43f1595d439296160/p/dbnav/journeys-req.js#L49 should fix this

### PR DESCRIPTION
**Pull request by dabund24**
resolves #5

### CODE REVIEW

File: p/dbnav/journeys-req.js [hunk 1]
```diff
@@ -46,7 +46,7 @@ const formatJourneysReq = (ctx, from, to, when, outFrwd, journeysRef) => {
 				? [{locationId: profile.formatLocation(profile, opt.via, 'opt.via').lid}]
 				: undefined,
 			zielLocationId: to.lid,
-			maxUmstiege: opt.transfers || undefined,
+			maxUmstiege: opt.transfers === -1 ? undefined : opt.transfers, // setting `maxUmstiege` to -1 is not allowed, see https://github.com/public-transport/db-vendo-client/issues/5
```
Comment (traines-source) [inline p/dbnav/journeys-req.js:49]: It's important that `opt.transfers` is set to `undefined (and not to `null) for the dbnav api. Maybe it's better to change that in `index.js`, given that nobody tries to pass `null`. I think this might also be what breaks the int tests (not the unit tests, that's because npmjs.org is apparently unreliable today)

Comment (dabund24) [reply p/dbnav/journeys-req.js:49]: Oh, makes sense, I'll change that.
However, I think `||` should be changed to `??` then, since `0` is also falsish and thus searching for direct connections only becomes impossible

**Comment by dabund24** [comment]
2535bb6 also removed the new unit test, which tested the `formatJourneysReq()` function of the db profile.
For re-adding a unit test for this pr, would you prefer the test to sit in its own file or should I just add it to `tests/db-journey.js`?

File: p/dbnav/journeys-req.js (hunk 1 above)
Comment (traines-source) [reply p/dbnav/journeys-req.js]: Ah yes good catch!

**Comment by traines-source** [comment]
Yes sad to see the unit test go, it also tested the 0/null thing :) (although I'm confused, was it ever green? or why did the `null` -> `undefined` work :thinking: ) 
Keep it where you prefer, although for the -1 thing it will probably be difficult to test with a unit test?

**Comment by dabund24** [comment]
It did in fact pass, but by mistake 😅
`same()` apparently allows conversion between `null` and `undefined`. I should have used `equal()` instead, where the test does not pass. Oh well...

**Comment by dabund24** [comment]
I could maybe add a second fixture where `transfers` is set to zero and then check for the correct result for different `transfers` values. What do you think?

**Comment by traines-source** [comment]
Ah good to know, but yeah now that I remember that in fact `null == undefined`, no wonder. And, well, `same()` is used throughout the codebase, so that's normal :)
Fixture, you mean in the e2e tests? The problem with these is that in CI they are also run against the live APIs, so we don't want too many requests. Or you meant introducing unit test fixtures for requests? That might be possible. I would probably just leave this untested tbh and just keep the null/0 unit test or move the -1 back into the journey-reqs again :see_no_evil: Do it however you see fit :)

**Comment by dabund24** [comment]
I think all in all, the best solution may even be to move this to a new file `format/journeys-req.js`, where option modifications concerning all profiles would live. This would allow for easy testing on the one hand and prevent code duplication on the other hand, so we'd kind of have the best of both worlds here. This would also ensure similar changes won't bloat `index.js` in the future

**Comment by traines-source** [comment]
If the potential next profile doesn't actually require transfers to be set to -1 :D These DB APIs are always great for surprises. And I have removed the hooking logic used in hafas-client for stacking transforms/formats. So not sure how exactly one would then elegantly call the individual profile-specific journey-reqs from there. Maybe one could just introduce a `formatTransfers` function that can be called from the two journey-reqs?

**Comment by dabund24** [comment]
That sounds like a good idea, I'll do that

**Comment by dabund24** [comment]
I *think* this is it (finally)

**Comment by dabund24** [comment]
wait, forgot removing prior changes to index.js

**Comment by traines-source** [comment]
Cool :) Thanks a lot!

**Comment by dabund24** [comment]
Thanks for the detailed feedback, genuinely improved my understanding of the codebase ^^
